lemonspotter/core/function.py

```diff
@@ -41,78 +47,134 @@ def name(self) -> str:
         """This property provides access to the Function name."""
 
         if 'name' not in self._json:
-            raise Exception('Function name is not in JSON.')
+            raise RuntimeError('Function name is not in JSON.')
 
         return self._json['name']
 
     @property
     def has_parameters(self) -> bool:
         """"""
 
-        return self._json.get('parameters', False)
+        return 'parameters' in self._json
+
+    @property
+    def has_in_parameters(self) -> bool:
+        """"""
+
+        return len(self.in_parameters) > 0
+
+    @property
+    def has_inout_parameters(self) -> bool:
+        """"""
+
+        return len(self.inout_parameters) > 0
+
+    @property
+    def has_out_parameters(self) -> bool:
+        """"""
+
+        return len(self.out_parameters) > 0
```

Comment:
Why does has_parameters check json, but all other "has_[something]_parameters" functions don't.

Reply:
The has_parameters is for if there are any. The others need to be more specific and that is not encoded in the JSON actually only per parameter. So they are forced to look for * parameters.